File: zig/scrabble-score/scrabble_score.zig
const std = @import("std");

pub fn score(s: []const u8) u32 {
    var sum: u32 = 0;

    for (s) |c| {
        sum += switch (std.ascii.toUpper(c)) {
            'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' => 1,
            'D', 'G' => 2,
            'B', 'C', 'M', 'P' => 3,
            'F', 'H', 'V', 'W', 'Y' => 4,
            'K' => 5,
            'J', 'X' => 8,
            'Q', 'Z' => 10,
            else => 0,
        };
    }

    return sum;
}

File: c/scrabble-score/scrabble_score.c
#include "scrabble_score.h"

#include <ctype.h>

int scores[26] = {1, 3, 3, 2, 1, 4, 2, 4, 1, 8, 5, 1, 3, 1, 1, 3, 10, 1, 1, 1, 1, 4, 4, 8, 4, 10};

unsigned int score(const char *word)
{
    unsigned int score = 0;

    while (*word)
    {
        score += scores[toupper(*word) - 'A'];
        word++;
    }

    return score;
}